src/Microsoft.Azure.WebJobs.Host/JobHost.cs

```diff
@@ -105,7 +105,7 @@ internal IListener Listener
         /// <summary>Starts the host.</summary>
         public void Start()
```

Comment:
@mathewc @mamaso this keeps coming back to mind. As we discussed, this is fine to get us unblocked and things up and running on with CI, but one conversation I'd like to get started is considering removal of **all** of this sync-over-async methods as they're always a source of issues, hide some important details from the caller and really provide very little benefit (a caller could still choose to make the async call if they wished and they'd be in control of how it's being scheduled, how continuations are setup, etc.).
Just planting the seed :)

/cc @MikeStall